proposals/accepted/50-ide-test-integration.md

```diff
@@ -0,0 +1,219 @@
+## Abstract
+
+This proposal seeks technical coordination from the Haskell Fondation
+for improving integration between IDEs and testing libraries in Haskell.
+
+This work will involve two major implementation areas: testing library
+integration, and IDE integration. The author will aim to implement both
+of these features, though help is certainly welcome :).
+
+This undertaking requires buy-in and feedback from the maintainers of
+Haskell's testing frameworks, since we would like to have one interface
+that satisfies everyone's needs.
+
+ ## Background
+
+IDE integration for test suites is a feature that is expected in most
+if not all major programming languages. First class support exists for
+[JUnit with multiple IDEs](https://junit.org/junit5/docs/current/user-guide/#running-tests-ide),
+[Python with VSC](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&cad=rja&uact=8&ved=2ahUKEwiRyt_VwcD7AhUXIUQIHQe1CjUQFnoECBIQAQ&url=https%3A%2F%2Fcode.visualstudio.com%2Fdocs%2Fpython%2Ftesting&usg=AOvVaw0U1x8oEhgUTX1GeoPJqVqo),
+[.NET with VSC](https://marketplace.visualstudio.com/items?itemName=formulahendry.dotnet-test-explorer)
+and many others.
+
+Often these IDEs will supply inline hints for "Run this test" and
+"Re-run failed tests" as well as others. For users of mainstream
+languages, such features are no longer features, but instead
+expectations from a mature language.
+
+Haskell has a mature and well-maintained testing ecosystem, including
+the libraries: `tasty`, `HUnit`, and `hspec`.  While these libraries
+themselves are mature, there is currently minimal
+integration between Haskell's testing libraries and HLS, as well as
+the primary IDEs used for Haskell (VSC;Emacs;Neovim).
+
+Attempts at integration between testing frameworks have focused on
+regex searches, with support for hspec:
+ 1. [neo-test](https://github.com/mrcjkb/neotest-haskell)
+ 2. [vim-test](https://github.com/vim-test/vim-test)
+
+While these are valuable contributions to the ecosystem, their choice
+of regex for the collection of the user's test tree, and support for only
```

Comment:
The term "test tree" is not necessarily understood by the audience for this proposal . Based on our discussion elsewhere, I think you made a good observation about how Haskell test libraries are typically (but not always) organized, and how it's different from something like JUnit-derived tests.  In particular, JUnit-inspired libraries typically use the language's notion of a declaration or definition to bound tests, so a test is a class or a method, while Haskell libraries more frequently use an ordinary data structure to represent tests. This means that we'd need to be a bit creative about the UI, and draw inspiration from things like IntelliJ while still doing something a bit different that fits Haskell better.

I think the proposal would be accessible to more readers if these assumptions and observations were made more explicit in this background section.